lib/AST/Decl.cpp

```diff
@@ -6046,15 +6046,79 @@ AnyFunctionType::Param ParamDecl::toFunctionParam(Type type) const {
   return AnyFunctionType::Param(type, label, flags);
 }
 
-void ParamDecl::setDefaultValue(Expr *E) {
+Initializer *ParamDecl::getDefaultArgumentInitContextCached() const {
```

Comment:
This could return an Optional<> I think. Also typically we name these methods "getCached...()"